addons/html_editor/static/src/main/link/link_plugin.js

```diff
@@ -605,7 +606,29 @@ export class LinkPlugin extends Plugin {
 
             const link = this.document.createElement("a");
             if (!selection.isCollapsed) {
-                const content = this.dependencies.selection.extractContent(selection);
+                const getFontSizeWrapper = (node) =>
+                    closestElement(
+                        node,
+                        (el) =>
+                            el.tagName === "SPAN" &&
+                            (FONT_SIZE_CLASSES.some((cls) => el.classList.contains(cls)) ||
+                                el.style?.fontSize)
+                    );
+
+                let content;
+                const fontSizeWrapper = getFontSizeWrapper(selection.commonAncestorContainer);
```

Comment:
Not sure what is the point of `getFontSizeWrapper` being a function because it is only used once, and its name does not provide any useful information that would not be already available from the code at call site.